uk.co.compucorp.civicrm.hrcore/hrcore.php

```diff
@@ -63,6 +63,24 @@ function hrcore_civicrm_container($container) {
   $loader->load('config/container/container.xml');
 }
 
+/**
+ * Implements hook_civicrm_buildForm
+ *
+ * @param string $formName
+ * @param object $form
+ */
+function hrcore_civicrm_buildForm($formName, &$form) {
+  if($formName == 'CRM_Activity_Form_ActivityFilter') {
+    $activityFilter = new CRM_HRCore_Hook_BuildForm_ActivityFilter();
+    $activityFilter->modifyExcludeAndIncludeActivityTypeFields($form);
+  }
+
+  if($formName == 'CRM_Activity_Form_ActivityLinks') {
+    $activityLinks = new CRM_HRCore_Hook_BuildForm_ActivityLinks();
+    $activityLinks->filterActivityTypeLinks($form);
+  }
+}
```

Comment:
Sorry I wasn't clear on my first comment, but I really meant you should follow the exact same approach as seen on `hook_civicrm_pre`. Note that there, we're transforming the hook in an event and the classes are listeners of this event, including being responsible to check if they should handle the event or not (check how the [PrimaryAddressSetter](https://github.com/civicrm/civihr/blob/staging/uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Hook/Pre/PrimaryAddressSetter.php) is implemented